horreum-api/src/main/java/io/hyperfoil/tools/horreum/api/internal/services/UserService.java

```diff
@@ -146,4 +170,35 @@ class NewUser {
         public String team;
         public List<String> roles;
     }
+
+    /**
+     * Key type allows to scope what the key gives access to
+     */
+    enum KeyType {
+        USER
+    }
+
+    class ApiKeyRequest {
+        public String name;
+        public KeyType type;
+
+        public ApiKeyRequest() {
+        }
+
+        public ApiKeyRequest(String name, KeyType type) {
+            this.name = name;
+            this.type = type;
+        }
+    }
+
+    class ApiKeyResponse {
+        public long id;
+        public String name;
+        public KeyType type;
+        public LocalDate creation;
+        public LocalDate access;
```

Comment:
I used `LocalDate` so that I don't have to go fine grain on the expiration. `Instant` represents a defined point in time but in this case I just want the key to be revoked ao some time after the expiration deadline.

AFAICT is encodes well to JSON (it's simply `[ yyyy, mm , dd ]`).

Reply:
Simple dates are always simple, until you start to try and use them ;)

if it encodes to `[ yyyy, mm , dd ]` does it decode to `[ yyyy, dd , mm ]`? does that depend on locale on machine doing the decoding, how does it port over to other clients, (go / python).  

Atm this in in the `internal` api, but the only reason that split exists is because we didn't have time to fix all the API portability issues, so only focused on the important ones for pushing data and pulling results.

This is adding a new date encoding to the API that we decided not to use because of the issues it casues

Reply:
makes sense. I was not in a disagreement, just giving context.